array/DNAm/preprocessing/checkColnamesFunction.r

```diff
@@ -0,0 +1,51 @@
+
+
+# Checks sample sheet columns are formatted correctly
+checkColnames <- function(sampleSheet, exp_cols, type, verbose=T){
+	metrics <- list()
+	
+	all_pres <- all(exp_cols %in% colnames(sampleSheet))
+	metrics[['allPresent']] <- all_pres
+	
+	if(all_pres){
+		if(verbose){print(paste(type, "column names: All present"))}
+	}else{
+	
+		if(verbose){print(paste(type, "column names: Not all present"))}
+		
+		#a) find correct cols
+		pres <- exp_cols[exp_cols %in% colnames(sampleSheet)]
+		if(length(pres)>0){
+			if(verbose){print(c("Identical colnames:",pres))}
+		}
+		metrics[['presentIDs']] <- pres
+		
+		#b) find absent cols
+		ab <- exp_cols[exp_cols %ni% colnames(sampleSheet)]
+		if(verbose){print(c("Absent colnames:",ab))}
+		metrics[['absentIDs']] <- ab
+		
+		#c) find close matches
+		near <- exp_cols[amatch(colnames(sampleSheet), exp_cols, maxDist = 2)] # maxDist = number of characters difference

```

Comment:
Is this `amatch` from the stringdist package? My version of R doesn't recognise this function and I don't see any namespaces being loaded.

Reply:
Also, it might be worth stripping `exp_cols` and `colnames(sampleSheet)` of any common separators (".", "_", "-" *etc.*) and making them all lower case first. Then you can be more stringent on `maxDist`. 

Currently the matching algorithm is set to "Damerau–Levenshtein distance" I believe. If you have any similarly named columns in any sampleSheets (never worked with one so this might not be an issue), then this might falsely match two columns.